frontend/src/services/session.service.ts

```diff
@@ -1,16 +1,71 @@
 import {Message} from "../models/message";
+import {Artwork} from "../models/artwork";
+import {Game} from "../models/game";
 
 export default class SessionService {
-    public static saveMessages(messages: Message[]): void {
-        localStorage.setItem('messages', JSON.stringify(messages));
+    private static saveInCache<T>(key: string, object: T): void {
+        SessionService.saveTimestamp();
+        localStorage.setItem(key, JSON.stringify(object));
+    }
+
+    private static getFromCache<T>(key: string): T | null {
+        if (SessionService.reloadRequired()) {
```

Comment:
I know this is bounded by process.env.REACT_APP_CACHE_LIFESPAN. Comment about how long this is would be nice.